kubernetes-addons/aws-node-termination-handler/main.tf

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT-0
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the "Software"), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+resource "aws_autoscaling_lifecycle_hook" "aws_node_termination_handler_hook" {
+  count = length(var.autoscaling_group_names)
+
+  name                   = "aws_node_termination_handler_hook"
+  autoscaling_group_name = var.autoscaling_group_names[count.index]
+  default_result         = "CONTINUE"
+  heartbeat_timeout      = 300
+  lifecycle_transition   = "autoscaling:EC2_INSTANCE_TERMINATING"
+}
+
+resource "aws_autoscaling_group_tag" "aws_node_termination_handler_tag" {
+  count = length(var.autoscaling_group_names)
+
+  autoscaling_group_name = var.autoscaling_group_names[count.index]
+
+  tag {
+    key   = "aws-node-termination-handler/managed"
+    value = "true"
+
+    propagate_at_launch = true
+  }
+}
+
+
+resource "aws_sqs_queue" "aws_node_termination_handler_queue" {
+  name_prefix               = "aws_node_termination_handler"
+  message_retention_seconds = "300"
+}
+
+resource "aws_sqs_queue_policy" "test" {
```

Comment:
`resource "aws_sqs_queue_policy" "test"` can be renamed to  `resource "aws_sqs_queue_policy" "handler_queue"`

Reply:
addressed